File: unittests/pytests/utils/testutils.py
#!/usr/bin/env nemesis
#
# ======================================================================
#
# Brad T. Aagaard, U.S. Geological Survey
# Charles A. Williams, GNS Science
# Matthew G. Knepley, University of Chicago
#
# This code was developed as part of the Computational Infrastructure
# for Geodynamics (http://geodynamics.org).
#
# Copyright (c) 2010-2017 University of California, Davis
#
# See COPYING for license information.
#
# ======================================================================
#

# @file unittests/utils/testutils.py

# @brief Python application for testing utils code.

from pylith.tests.UnitTestApp import UnitTestApp

import unittest


class TestApp(UnitTestApp):
    """
    Test application.
    """

    def __init__(self):
        """
        Constructor.
        """
        UnitTestApp.__init__(self)
        return

    def _suite(self):
        """
        Setup the test suite.
        """

        suite = unittest.TestSuite()

        from TestPylith import TestPylith
        suite.addTest(unittest.makeSuite(TestPylith))

        from TestEventLogger import TestEventLogger
        suite.addTest(unittest.makeSuite(TestEventLogger))

        from TestPylithVersion import TestPylithVersion
        suite.addTest(unittest.makeSuite(TestPylithVersion))

        from TestPetscVersion import TestPetscVersion
        suite.addTest(unittest.makeSuite(TestPetscVersion))

        from TestDependenciesVersion import TestDependenciesVersion
        suite.addTest(unittest.makeSuite(TestDependenciesVersion))

        from TestCollectVersionInfo import TestCollectVersionInfo
        suite.addTest(unittest.makeSuite(TestCollectVersionInfo))

        from TestConstants import TestConstants
        suite.addTest(unittest.makeSuite(TestConstants))

        return suite


# ----------------------------------------------------------------------
if __name__ == '__main__':
    app = TestApp()
    app.run()


# End of file

File: unittests/libtests/utils/test_utils.cc
// -*- C++ -*-
//
// ----------------------------------------------------------------------
//
// Brad T. Aagaard, U.S. Geological Survey
// Charles A. Williams, GNS Science
// Matthew G. Knepley, University of Chicago
//
// This code was developed as part of the Computational Infrastructure
// for Geodynamics (http://geodynamics.org).
//
// Copyright (c) 2010-2015 University of California, Davis
//
// See COPYING for license information.
//
// ----------------------------------------------------------------------
//

#include "petsc.h"

#include <cppunit/extensions/TestFactoryRegistry.h>

#include <cppunit/BriefTestProgressListener.h>
#include <cppunit/extensions/TestFactoryRegistry.h>
#include <cppunit/TestResult.h>
#include <cppunit/TestResultCollector.h>
#include <cppunit/TestRunner.h>
#include <cppunit/TextOutputter.h>

#include <stdlib.h> // USES abort()

int
main(int argc,
     char* argv[])
{ // main
  CppUnit::TestResultCollector result;

  try {
    // Initialize PETSc
    PetscErrorCode err = PetscInitialize(&argc, &argv, NULL, NULL);CHKERRQ(err);
    err = PetscOptionsSetValue(NULL, "-malloc_dump", "");CHKERRQ(err);

    // Create event manager and test controller
    CppUnit::TestResult controller;

    // Add listener to collect test results
    controller.addListener(&result);

    // Add listener to show progress as tests run
    CppUnit::BriefTestProgressListener progress;
    controller.addListener(&progress);

    // Add top suite to test runner
    CppUnit::TestRunner runner;
    runner.addTest(CppUnit::TestFactoryRegistry::getRegistry().makeTest());
    runner.run(controller);

    // Print tests
    CppUnit::TextOutputter outputter(&result, std::cerr);
    outputter.write();

    // Finalize PETSc
    err = PetscFinalize();
    CHKERRQ(err);
  } catch (...) {
    abort();
  } // catch

  return (result.wasSuccessful() ? 0 : 1);
} // main


// End of file
